### ISSUE TITLE
Allow Mondrian Forest to continue learning from an existing Ensemble Model

### ISSUE THREAD

**Issue by NoRaincheck** [NoRaincheck/shinrin#4]
# Idea

Assume there is a gradient boosting model or random forest model from `onnx`, import the tree to a mondrian tree and continue training. 

Note that if we do this we need to build the statistics up again. We take a k-NN style approach (or similar to TabICL) where the signature is something like:

```py
model = MondrianForest.from_model(...)

# requires data set to be sufficient large to regenerate the statistics to move it to mondrian model, e.g. at least 300 rows
# this shouldn't change the model at all. It just updates statistics so it works with Mondrian Forest
# there should be something here to indicate that it hasn't been fitted or its from a model and therefore we're building statistics only
# or an optionla parameter to do it explicitly, like build_mondrian_process_statistics or something more succinct
model.fit(X, y)  

# afterwards you can use partial fit as normal
model.partial_fit(X_partial, y_partial)
```

Note that we shouldn't 'break' existing APIs. That is 

```py
model.fit(X, y)
```

should still completely refit a model if its not in some mode where it is finding statistics. We should be clear in our warnings and messages that we are building statistics etc. The API above is more 'convenience' and assumes 'happy path'. Ideally user is always explicit in what they are doing.
